src/Algebra/Properties/Monoid/Reasoning.agda

```diff
@@ -0,0 +1,70 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Equational reasoning for monoids
+-- (Utilities for identity and cancellation reasoning, extending semigroup reasoning)
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+open import Algebra.Bundles using (Monoid)
+
+module Algebra.Properties.Monoid.Reasoning {o ℓ} (M : Monoid o ℓ) where
+
+open Monoid M
+  using (Carrier; _∙_; _≈_; setoid; isMagma; semigroup; ε; sym; identityˡ
+  ; identityʳ ; ∙-cong; refl; assoc; ∙-congˡ; ∙-congʳ; trans)
+open import Relation.Binary.Reasoning.Setoid setoid
+
+open import Algebra.Properties.Semigroup semigroup public
+
+private
+    variable
+        a b c d : Carrier
+
+module _ where
+    id-unique : ∀ a → (∀ b → b ∙ a ≈ b) → a ≈ ε
+    id-unique a b∙a≈b = trans (sym (identityˡ a)) (b∙a≈b ε)
+
+    id-comm : ∀ a → a ∙ ε ≈ ε ∙ a
+    id-comm a = trans (identityʳ a) (sym (identityˡ a))
+
+    id-comm-sym : ∀ a → ε ∙ a ≈ a ∙ ε
+    id-comm-sym a = sym (id-comm a)
+
+module _ {a b : Carrier} (a≈ε : a ≈ ε) where
+    elimʳ : ∀ b → b ∙ a ≈ b
+    elimʳ b = trans (∙-congˡ a≈ε) (identityʳ b)
+
+    elimˡ : ∀ b → a ∙ b ≈ b
+    elimˡ b = trans (∙-congʳ a≈ε) (identityˡ b)
+
+    introʳ : ∀ b → b ≈ b ∙ a
+    introʳ b = sym (elimʳ b)
+
+    introˡ : ∀ b → b ≈ a ∙ b
+    introˡ b = sym (elimˡ b)
+
+    introcenter : ∀ c → b ∙ c ≈ b ∙ (a ∙ c)
+    introcenter c = trans (∙-congˡ (sym (identityˡ c))) (∙-congˡ (∙-congʳ (sym a≈ε)))
```

Comment:
I wonder if we should use a superscript `m` (for 'middle') or `c` (for, indeed, 'center') instead of spelling it out?

Reply:
Meta: a good general point that, besides 'left' and 'right' (modulo collective mutual misunderstanding/inconsistency about what such conventions might even mean #2654 ), we don't have good heuristics about super-/sub-scripts and their intended semantics. I *think* I'd be happy with `ᶜ` or `ᵐ`... but nervous about establishing a precedent without further thought/discussion?

Reply:
On balance, and with `cancelInner` below, I now think I'd prefer the superscript version, but don't mind which, maybe a slight preference for 'c' (possible Anglophone/Francophone bias?)